src/app/components/DataLoader/DataLoader.jsx

```diff
@@ -1,10 +1,73 @@
 import React from 'react';
+import PropTypes from 'prop-types';
+import Actions from '@Actions';
+import { ajaxCall } from '@utils';
+import appConfig from '@appConfig';
 
 class DataLoader extends React.Component {
+  constructor(props) {
+    super(props);
+    const { location } = this.props;
+    this.pathInstructions = [
+      {
+        expression: /\/research\/collections\/shared-collection-catalog\/bib\/(b\d*)/,
+        pathType: 'bib',
+      },
+    ];
+    this.pathType = null;
+    this.routes = {
+      bib: {
+        apiRoute: () => `${appConfig.baseUrl}/api/bib?bibId=${this.matchData[1]}`,
```

Comment:
The structure of the object seems fine, but it does worry me that `this.matchData` is not defined until the component is mounted. This works because you're using it in a function, but it's hard to follow the flow of when it's assigned a value in a different function.

I would make the `bibId` a param of the function and pass it in when you call the function. So something like
```
apiRoute: (bibId) => `${appConfig.baseUrl}/api/bib?bibId=${bibId}`,
```

And then call it in `componentDidMount`:
```
const matchData = this.pathInstructions.reduce(this.reducePathExpressions, null);
// ...
  ajaxCall(apiRoute(matchData[1]),
// ...
```

That way, you also remove the need to have the component instance variable of `this.matchData`, and the `apiRoute` function can be slightly more generic.

Reply:
Good point about making `matchData` a local variable of the function instead of an instance attribute. I've implemented a slight modification of your suggestion